app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -1,37 +1,50 @@
-ManageIQ.angular.app.controller('cloudSubnetFormController', ['$scope', 'cloudSubnetFormId', 'miqService', 'API', function($scope, cloudSubnetFormId, miqService, API) {
-  $scope.cloudSubnetModel = { name: '', ems_id: '', cloud_tenant_id: '', network_id: '' };
-  $scope.formId = cloudSubnetFormId;
-  $scope.afterGet = false;
-  $scope.modelCopy = angular.copy( $scope.cloudSubnetModel );
-  $scope.model = "cloudSubnetModel";
+ManageIQ.angular.app.controller('cloudSubnetFormController', ['cloudSubnetFormId', 'miqService', 'API', function(cloudSubnetFormId, miqService, API) {
+  var vm = this;
 
-  ManageIQ.angular.scope = $scope;
+  var init = function() {
+    vm.afterGet = false;
+    vm.cloudSubnetModel = {
+      name: "",
+      ems_id: "",
+      cloud_tenant_id: "",
+      network_id: "",
+    };
 
-  if (cloudSubnetFormId == 'new') {
-    $scope.cloudSubnetModel.name = "";
-    $scope.cloudSubnetModel.dhcp_enabled = true;
-    $scope.cloudSubnetModel.network_protocol = '4';
-    $scope.newRecord = true;
-  } else {
-    miqService.sparkleOn();
-    API.get("/api/cloud_subnets/" + cloudSubnetFormId + "?expand=resources").then(function(data) {
-      $scope.afterGet = true;
-      $scope.cloudSubnetModel.name = data.name;
-      $scope.cloudSubnetModel.dhcp_enabled = data.dhcp_enabled;
-      $scope.cloudSubnetModel.cidr = data.cidr;
-      $scope.cloudSubnetModel.gateway = data.gateway;
-      $scope.cloudSubnetModel.network_protocol = data.network_protocol;
-      $scope.modelCopy = angular.copy( $scope.cloudSubnetModel );
+    vm.networkProtocols = ["ipv4", "ipv6"];
+    vm.formId = cloudSubnetFormId;
+    vm.model = "cloudSubnetModel";
+    vm.newRecord = cloudSubnetFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    if (vm.newRecord) {
+      vm.cloudSubnetModel.dhcp_enabled = true;
+      vm.cloudSubnetModel.network_protocol = "ipv4";
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.cloudSubnetModel );
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/cloud_subnets/" + cloudSubnetFormId + "?expand=resources&attributes=ext_management_system.name,cloud_tenant.name,cloud_network.name").then(function(data) {
+        vm.cloudSubnetModel.name = data.name;
+        vm.cloudSubnetModel.ems_name = data.ext_management_system.name;
+        vm.cloudSubnetModel.tenant_name = angular.isDefined(data.cloud_tenant) ? data.cloud_tenant.name : undefined;
```

Comment:
Well, if you *need* to check for definedness, use `data.cloud_tenant !== undefined`. 

But since you just return undefined when cloud tenant is undefined, you don't need to check that.

This could also be written as `data.cloud_tenant && data.cloud_tenant.name`.